addons/hr_recruitment/static/src/scss/hr_applicant.scss

```diff
@@ -19,3 +19,7 @@
         }
     }
 }
+
+.o_field_phone {
+    word-break: keep-all !important;
```

Comment:
As @aab-odoo said, this should probably be moved to `web/` in [phone_field.scss]

[phone_field.scss]: https://github.com/odoo/odoo/blob/16.0/addons/web/static/src/views/fields/phone/phone_field.scss